### ISSUE TITLE
MIUI ActivityInjector: get life cycle exception

### ISSUE THREAD

**Issue by flamyoad** [flamyoad/Honnoki#38]
[MIUI 11/12 Theme Switch Results in LifeCycleException, ClassCastException](https://stackoverflow.com/questions/64565156/miui-11-12-theme-switch-results-in-lifecycleexception-classcastexception)

[Getting Lifecycle Exception While Recreating The Activity](https://stackoverflow.com/questions/63209993/getting-lifecycle-exception-while-recreating-the-activity)

```
2022-04-09 18:24:45.459 3653-3653/com.flamyoad.honnoki.debug E/ActivityInjector: get life cycle exception
    java.lang.ClassCastException: android.os.BinderProxy cannot be cast to android.app.servertransaction.ClientTransaction
        at android.app.ActivityInjector.checkAccessControl(ActivityInjector.java:24)
        at android.app.Activity.onResume(Activity.java:2036)
        at androidx.fragment.app.FragmentActivity.onResume(FragmentActivity.java:433)
        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1457)
        at android.app.Activity.performResume(Activity.java:8407)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4602)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4644)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ClientTransactionHandler.executeTransaction(ClientTransactionHandler.java:58)
        at android.app.ActivityThread.handleRelaunchActivityLocally(ActivityThread.java:5617)
        at android.app.ActivityThread.access$3500(ActivityThread.java:265)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2178)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:236)
        at android.app.ActivityThread.main(ActivityThread.java:8168)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
```

No issue on my `POCO X3 Pro` (MIUI 12.0.3, Android 11). But crashes on `Xiaomi Pad 5` (MIUI 13, Android 11).

Anyway, still don't understand why. Maybe POCO has added its own modification on top of MIUI itself.

**Comment by flamyoad** [comment]
https://github.com/flamyoad/Honnoki/blob/3e2808078d1f7df8f73f13557f3322f877110943/app/src/main/java/com/flamyoad/honnoki/ui/reader/ReaderActivity.kt#L231-L234

**Comment by 002gnahz** [comment]
I'm having this problem too, so have you found a solution, if so please share it with me！Thanks.

**Comment by flamyoad** [comment]
Hello @002gnahz . Apparently in my case, it was not MIUI's fault. It was caused by incorrect use of Koin that somehow escalated to `MIUI ActivityInjector: get life cycle exception`. I injected the wrong type of object.

This is my commit that fixed the issue https://github.com/flamyoad/Honnoki/commit/a66d86006320734db5e0e35f5ea7eab9a4687afb

However, it should not be related to your issue. My advise is that you check all possible sources of runtime exceptions because they may lead to the MIUI exception above.
